### PR TITLE
[repo] cleanup #nullable disable

### PR DESCRIPTION
**Pull request by Kielek**
Towards #3958

Please merge after #5900

## Changes

[repo] cleanup #nullable disable + update contributing guide

There is couple (3) of leftovers of `#nullable disable`. It is needed for auto-generated files.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* ~~[ ] Unit tests added/updated~~
* ~~[ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes~~
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

**Review by ysolomchenko — APPROVED** [review]
LGTM

**Review by CodeBlanch — APPROVED** [review]
LGTM

File: CONTRIBUTING.md
Comment (CodeBlanch) [inline CONTRIBUTING.md:1]: @Kielek 

Should we remove "nullable enabled" from here too?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/1a68b49ed61710ea1e334f5fd8c818cc687a96ae/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L10

Comment (ysolomchenko) [reply CONTRIBUTING.md:1]: I think we can leave it there for new contributors who create projects and files with nullable disabled for some reason, and they can already see this line where they can read and understand that files must have nullable enabled.

**Pull request merged**